haystack/nodes/ranker/most_diverse.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from pathlib import Path
+from typing import List, Literal, Optional, Union
+
+from haystack.nodes import BaseRanker
+from haystack.schema import Document
+from haystack.lazy_imports import LazyImport
+
+logger = logging.getLogger(__name__)
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as torch_and_transformers_import:
+    import torch
+    from sentence_transformers import SentenceTransformer
+    from haystack.modeling.utils import initialize_device_settings  # pylint: disable=ungrouped-imports
+
+
+class MostDiverseRanker(BaseRanker):
+    """
+    Implements a document ranking algorithm that orders documents in such a way as to maximize the overall diversity
+    of the documents.
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: Union[str, Path] = "all-MiniLM-L6-v2",
+        use_gpu: Optional[bool] = True,
```

Comment:
I just noticed that the type hint is Optional[bool] which means that you can also set use_gpu to None. Is there a reason for this? It's not clear to me what the difference between use_gpu=False and use_gpu=None would be.

Reply:
I think that's mostly due to legacy we have in `initialize_device_settings`